includes/alma-wc-functions.php

```diff
@@ -63,37 +76,70 @@ function alma_wc_array_merge_recursive() {
 	return $merged;
 }
 
+/**
+ * Get eligible installments for price according to settings.
+ *
+ * @param int $amount the amount to pay.
+ *
+ * @return int[]
+ */
+function alma_wc_get_eligible_installments_according_to_settings( $amount ) {
+	$allowed_installments_list  = alma_wc_plugin()->settings->get_enabled_pnx_plans_list();
+	$eligible_installments_list = array();
+
+	foreach ( $allowed_installments_list as $plan ) {
+		if ( $amount >= $plan['min_amount'] && $amount <= $plan['max_amount'] ) {
+			$eligible_installments_list[] = $plan['installments'];
+		}
+	}
+
+	return $eligible_installments_list;
+}
```

Comment:
A minima je pense qu'on peut clarifier l'implem de cette fonction avec des noms de variables plus courts (i.e. le pluriel et `_list` sont redondants).

De plus, je pense que le `according_to_settings` dans le nom de la fonction nous dit clairement que cette fonction devrait plutôt être une méthode `get_eligible_installments` sur la classe `WC_Alma_Settings` 🙂 

```suggestion
function alma_wc_get_eligible_installments_according_to_settings( $amount ) {
	$allowed_installments  = alma_wc_plugin()->settings->get_enabled_pnx_plans();
	$eligible_installments = array();

	foreach ( $allowed_installments as $plan ) {
		if ( $amount >= $plan['min_amount'] && $amount <= $plan['max_amount'] ) {
			$eligible_installments[] = $plan['installments'];
		}
	}

	return $eligible_installments;
}
```